src/main/scala/sangria/slowlog/OpenTracing.scala

```diff
@@ -74,7 +80,10 @@ class OpenTracing(parentSpan: Option[Span] = None, defaultOperationName: String
       error: Throwable,
       mctx: MiddlewareQueryContext[Any, _, _],
       ctx: Context[Any, _]) =
-    queryVal.get(ctx.path.path).foreach(_.finish())
+    queryVal.get(ctx.path.path).foreach(scope => {
+      scope.span().finish()
+      scope.close()
+    })
 }
 
-final case class SpanAttachment(span: Span) extends MiddlewareAttachment
\ No newline at end of file
+final case class ScopeAttachment(span: Scope) extends MiddlewareAttachment
```

Comment:
Maybe we can also add a helper to directly get the `span`. WDYT? For example: 

```scala
def span = scope.span
```

We probably also should rename `span: Scope` → `scope: Scope`